pkg/services/ssosettings/strategies/oauth_strategy.go

```diff
@@ -32,26 +32,29 @@ func (s *OAuthStrategy) IsMatch(provider string) bool {
 	section := s.cfg.SectionWithEnvOverrides("auth." + s.provider)
 
+	// TODO: load the provider specific keys separately
 	result := map[string]interface{}{
-		"client_id":                  section.Key("client_id").Value(),
-		"client_secret":              section.Key("client_secret").Value(),
-		"scopes":                     section.Key("scopes").Value(),
-		"auth_url":                   section.Key("auth_url").Value(),
-		"token_url":                  section.Key("token_url").Value(),
-		"api_url":                    section.Key("api_url").Value(),
-		"teams_url":                  section.Key("teams_url").Value(),
-		"enabled":                    section.Key("enabled").MustBool(false),
-		"email_attribute_name":       section.Key("email_attribute_name").Value(),
-		"email_attribute_path":       section.Key("email_attribute_path").Value(),
-		"role_attribute_path":        section.Key("role_attribute_path").Value(),
-		"role_attribute_strict":      section.Key("role_attribute_strict").MustBool(false),
-		"groups_attribute_path":      section.Key("groups_attribute_path").Value(),
-		"team_ids_attribute_path":    section.Key("team_ids_attribute_path").Value(),
-		"allowed_domains":            section.Key("allowed_domains").Value(),
-		"hosted_domain":              section.Key("hosted_domain").Value(),
-		"allow_sign_up":              section.Key("allow_sign_up").MustBool(true),
-		"name":                       section.Key("name").MustString("default name"), // TODO: change this default value
-		"icon":                       section.Key("icon").Value(),
+		"client_id":               section.Key("client_id").Value(),
+		"client_secret":           section.Key("client_secret").Value(),
+		"scopes":                  section.Key("scopes").Value(),
+		"auth_url":                section.Key("auth_url").Value(),
+		"token_url":               section.Key("token_url").Value(),
+		"api_url":                 section.Key("api_url").Value(),
+		"teams_url":               section.Key("teams_url").Value(),
+		"enabled":                 section.Key("enabled").MustBool(false),
+		"email_attribute_name":    section.Key("email_attribute_name").Value(),
+		"email_attribute_path":    section.Key("email_attribute_path").Value(),
+		"role_attribute_path":     section.Key("role_attribute_path").Value(),
+		"role_attribute_strict":   section.Key("role_attribute_strict").MustBool(false),
+		"groups_attribute_path":   section.Key("groups_attribute_path").Value(),
+		"team_ids_attribute_path": section.Key("team_ids_attribute_path").Value(),
+		"allowed_domains":         section.Key("allowed_domains").Value(),
+		"hosted_domain":           section.Key("hosted_domain").Value(),
+		"allow_sign_up":           section.Key("allow_sign_up").MustBool(true),
+		"name":                    section.Key("name").MustString("default name"), // TODO: change this default value
+		"icon":                    section.Key("icon").Value(),
```

Comment:
There are 3 similar mappings like this in the PR. It feels error prone since it's easy to introduce a typo. For example, the magic string `email_attribute_path` appears 4 times in this PR alone. Can we make it more generic?

Reply:
I am pretty sure loadOAuthInfo will be removed once all other providers have been updated cc @mgyongyosi

Reply:
Correct, the main plan was to remove the `loadOAuthInfo` at the end of these changes, however probably it can be removed now and we can use the other method which has been introduced for loading/mapping the data from the ini (I'm gonna add a mapping test to it just to be sure) . On the other hand, the `OAuthStrategy` implementation is different (it doesn't use the ini file), so we need that to be separate. At the end of the sso config development there will be only one implementation which will be in the OAuthStrategy.